htdocs/core/db/DoliDB.class.php

```diff
@@ -125,15 +125,20 @@ public function lasterrno()
 	 *
 	 * @param   string 	$stringtosanitize 	String to escape
 	 * @param   int		$allowsimplequote 	1=Allow simple quotes in string. When string is used as a list of SQL string ('aa', 'bb', ...)
+	 * @param   int		$allowspaces 		1=Allow spaces in string. When string is a sentense or a composed word
 	 * @return  string                      String escaped
 	 */
-	public function sanitize($stringtosanitize, $allowsimplequote = 0)
+	public function sanitize($stringtosanitize, $allowsimplequote = 0, $allowspaces = 0)
```

Comment:
The parameter $allowsspace was introduced in version v19 with signature
$allowsspace = 0

As it is a fix already fixed into a higher version, it must be fixed by doing a backport, so it means the fix must b done in a similar way: so parameter and var name must be same at same place (so must be $allowsspace and must be at position 4 in function signature)
So signature must be
```
public function sanitize($stringtosanitize, $allowsimplequote = 0, $allowsequals = 0, $allowsspace = 0)
```
even if param allowsequals is not used.

Because if you do not follow same signature, if you introduce the function differently, imagine someone, in a module is using it when he will upgrade to a newer version of dolibarr, his module will be broken.

Reply:
for sure !

Reply:
@eldy job is done : 

- sanitize function backported from dolibarr 19.0
- change call of db->sanitize used with right param orders